File: platform/scripts/_ai_model.py
"""Source de vérité UNIQUE du modèle Claude pour tous les scripts Python.

Pour changer la version de Sonnet (ou tester Opus/Haiku), il suffit de modifier
la valeur par défaut ci-dessous : ça s'applique PARTOUT d'un coup — génération
des classements (enrich_editorial), articles de blog (blog_publish_scheduled) et
avis (avis_publish_scheduled). Plus besoin d'éditer chaque script un par un.

Surchargeable à l'exécution sans toucher au code via la variable d'environnement
CLAUDE_MODEL (ex. pour tester un modèle ponctuellement).

Historique : "claude-sonnet-4-20250514" (Sonnet 4) a été retiré par Anthropic
(→ HTTP 404 not_found_error sur chaque appel). Migré vers "claude-sonnet-4-6".
"""
import os

# ⇩⇩⇩  LA SEULE LIGNE À CHANGER POUR METTRE À JOUR LE MODÈLE PARTOUT  ⇩⇩⇩
CLAUDE_MODEL = os.environ.get("CLAUDE_MODEL", "claude-sonnet-4-6")

File: saas/dashboard/lib/ai-model.ts
/**
 * Source de vérité UNIQUE du modèle Claude pour le dashboard.
 *
 * Pour changer la version de Sonnet (ou tester Opus/Haiku), modifier la valeur
 * par défaut ci-dessous : ça s'applique à TOUTES les routes API d'un coup
 * (blog, blog-meta, outils, codes-promo, regenerate-block, generate-text…).
 * Plus besoin d'éditer chaque route une par une.
 *
 * Surchargeable via la variable d'environnement CLAUDE_MODEL (Vercel) sans
 * toucher au code.
 *
 * Historique : "claude-sonnet-4-20250514" (Sonnet 4) retiré par Anthropic le
 * 15 juin 2026 (→ HTTP 404). Migré vers "claude-sonnet-4-6".
 */

// ⇩⇩⇩  LA SEULE LIGNE À CHANGER POUR METTRE À JOUR LE MODÈLE PARTOUT  ⇩⇩⇩
export const CLAUDE_MODEL = process.env.CLAUDE_MODEL || 'claude-sonnet-4-6';
